MediaGalleryUi/view/adminhtml/web/js/grid/masonry.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+define([
+    'Magento_Ui/js/grid/masonry',
+    'jquery'
+], function (Masonry, $) {
+    'use strict';
+
+    return Masonry.extend({
+        defaults: {
+            modules: {
+                provider: '${ $.provider }'
+            }
+        },
+
+        /**
+         * Init component
+         *
+         * @return {Object}
+         */
+        initialize: function () {
+            this._super();
+            this.initEvents();
+
+            return this;
+        },
+
+        /**
+         * Initialize events
+         */
+        initEvents: function () {
+            $(window).on('enhanced_media_gallery_folder_deleted', this.reloadGrid.bind(this));
```

Comment:
Let's rename the event for consistency with other events naming
```suggestion
            $(window).on('folderDeleted.enhancedMediaGallery', this.reloadGrid.bind(this));
```